test/unit/sdk.test.js

```diff
@@ -380,15 +363,157 @@ describe('sdk', () => {
         origin: CDN_ORIGIN,
       };
 
-      smartcar.messageHandler(evnt);
+      smartcar.messageHandler(event);
 
-      expect(smartcar.onComplete).toBeCalledWith(
-        null,
-        'super-secret-code',
-        'some-state'
-      );
+      expect(smartcar.onComplete).toBeCalledWith(null, 'super-secret-code', 'some-state');
     });
 
+    test(// eslint-disable-next-line max-len
+      'fires onComplete w/ VehicleIncompatible error when `error: vehicle_incompatible` in postMessage', () => {
+        const options = {
+          clientId: 'clientId',
+          redirectUri: `${CDN_ORIGIN}?app_origin=https://app.com`,
+          scope: ['read_vehicle_info', 'read_odometer'],
+          // eslint-disable-next-line no-unused-vars, no-empty-function
+          onComplete: jest.fn((__, _) => {}),
+        };
+
+        const smartcar = new Smartcar(options);
+        const errorDescription = 'describes the error';
+
+        const vehicleInfo = {
+          vin: 'some_vin',
+          year: 2017,
+          make: 'TESLA',
+          model: 'Model S',
+        };
+
+        const event = {
+          data: {
+            name: 'SmartcarAuthMessage',
+            isSmartcarHosted: true,
+            code: 'super-secret-code',
+            error: 'vehicle_incompatible',
+            errorDescription,
+            state: 'some-state',
+            ...vehicleInfo,
+          },
+          origin: CDN_ORIGIN,
+        };
+
+        smartcar.messageHandler(event);
+
+        expect(smartcar.onComplete).toBeCalledWith(
+          new Smartcar.VehicleIncompatible(errorDescription, vehicleInfo),
+          'super-secret-code',
+          'some-state',
+        );
+      });
+
+    test(// eslint-disable-next-line max-len
+      'VehicleIncompatible error does not add undefined properties to vehicleInfo', () => {
+        const options = {
+          clientId: 'clientId',
+          redirectUri: `${CDN_ORIGIN}?app_origin=https://app.com`,
+          scope: ['read_vehicle_info', 'read_odometer'],
+          // eslint-disable-next-line no-unused-vars, no-empty-function
+          onComplete(err, code, state) {
```

Comment:
nit: why not use a mock function here? That way all of the assertions can be at the bottom of the test, I initially thought this test was missing asserts

Reply:
Fixed